packages/inventory/src/components/table/InventoryList.js

```diff
@@ -11,87 +11,87 @@ import AccessDenied from '../../shared/AccessDenied';
+    const prevItems = useRef(props.items);
+    const prevSortBy = useRef(props.sortBy);
+
+    useEffect(() => {
+        if (props.hasItems) {
+            onRefreshData();
+        }
+    }, []);
 
     /**
      * Function to calculate for new changes, this function limits re-renders by checking if previous items are
      * same as new items.
      * If items are not passed, it only checks for props sortBy.
      * @param {*} prevProps previous props - items, hasItems, sortBy.
      */
-    componentDidUpdate(prevProps) {
-        const { items, hasItems, sortBy } = this.props;
-        if (
-            hasItems &&
-            !isEqual(
-                items.map(({ children, isOpen, ...item }) => item),
-                prevProps.items.map(({ children, isOpen, ...item }) => item)
-            )
-        ) {
-            this.onRefreshData({});
-        } else if (!hasItems && !isEqual(prevProps.sortBy, sortBy)) {
-            this.onRefreshData({});
+    useEffect(() => {
+        if (props.hasItems && !isEqual(prevItems.current, props.items)) {
+            prevItems.current = props.items;
+            onRefreshData();
         }
-    }
 
-    componentDidMount() {
-        if (this.props.hasItems) {
-            this.onRefreshData({});
+        if (!props.hasItems && !isEqual(prevSortBy.current, props.sortBy)) {
+            prevSortBy.current = props.sortBy;
+            onRefreshData();
         }
-    }
+    });
 
-    render() {
-        const { showHealth, ...props } = this.props;
-        return (
-            <Grid gutter="sm" className="ins-inventory-list">
-                <GridItem span={ 12 }>
-                    <InventoryEntityTable { ...props } />
-                </GridItem>
-            </Grid>
-        );
-    }
-}
+    return (<Grid gutter="sm" className="ins-inventory-list">
+        <GridItem span={ 12 }>
+            <InventoryEntityTable { ...props } />
+        </GridItem>
+    </Grid>);
+};
 
 /**
  * Component that consumes active filters and passes them down to component.
  */
 const InventoryList = React.forwardRef(({ hasAccess, getEntities, hideFilters, ...props }, ref) => {
     const dispatch = useDispatch();
     const activeFilters = useSelector(({ entities: { activeFilters } }) => activeFilters);
+
+    const loadEntities = (config, showTags) => dispatch(loadSystems({ ...config, hideFilters }, showTags, getEntities));
+
+    /**
+     * If conumer wants to change data they can call this function via component ref.
+     * @param {*} options new options to be applied, like pagination, filters, etc.
+     */
+    const onRefreshData = (options = {}) => {
+        const { page, perPage, items, hasItems, sortBy, activeFilters, showTags, customFilters } = props;
+        loadEntities({
+            page,
+            perPage,
+            items,
+            hasItems,
+            sortBy,
+            activeFilters,
+            ...customFilters,
+            ...options
+        }, showTags);
+    };
```

Comment:
Might be reasonable to combine these 2 functions together. The `loadEntities` function is not used anywhere else.